client/client.go

```diff
@@ -1131,6 +1151,74 @@ func (c *Client) run() {
 	}
 }
 
+// submitNodeEvents is used to submit a client-side node event. Examples of
+// these kinds of events include when a driver moves from healthy to unhealhty
+// (and vice versa)
+func (c *Client) submitNodeEvents(e []*structs.NodeEvent) error {
+	node := c.Node()
+	nodeEvents := map[string][]*structs.NodeEvent{
+		node.ID: e,
+	}
+	req := structs.EmitNodeEventRequest{
+		NodeEvents:   nodeEvents,
+		WriteRequest: structs.WriteRequest{Region: c.Region()},
+	}
+	var resp structs.EmitNodeEventResponse
+	if err := c.RPC("Node.EmitEvent", &req, &resp); err != nil {
+		c.logger.Printf("[ERR] client: emitting node events failed %v", err)
+		return err
+	}
+	c.logger.Printf("[INFO] client: emit node events complete")
+	return nil
+}
+
+// emitEvent is a handler which receives node events and on a interval and
+// submits them in batch format to the server
+func (c *Client) watchEmitEvents() {
+	batchEventsLock := sync.Mutex{}
+	batchEvents := make([]*structs.NodeEvent, 0)
+
+	timer := time.NewTimer(c.retryIntv(nodeEventsEmitIntv))
+	defer timer.Stop()
+
+	for {
+		select {
+		case event := <-c.triggerEmitNodeEvent:
+			batchEventsLock.Lock()
```

Comment:
This is more of a Go question - why does this need a lock if only one of these goroutines run per client, and this goroutine is the only one that owns appending to batchevents?

Reply:
+1 remove the lock